lpm/src/prefix/with_ports.rs

```diff
@@ -0,0 +1,299 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
Where are the tests for this new functionality?  I prefer fuzz tests if they are easy, but there should at least be some tests right?

Reply:
I'm not sure I see what functionality you refer to, in particular. There are some unit tests in follow-up commits after the introduction of these structs and methods.

Fuzzing is tricky, most of the interesting methods to fuzz require re-implementing the logic in the fuzzer to validate correctness. I don't have any fuzzing tests in this PR, we could look into it but it would probably be for a follow-up.

Reply:
I think the rest of the tests test this functionality enough to omit tests here for now.